pkg/resourceinterpreter/configurableinterpreter/configurable.go

```diff
@@ -2,142 +2,212 @@ package configurableinterpreter
-func (c *ConfigurableInterpreter) CheckConfigurableExists(gvk schema.GroupVersionKind) bool {
-	_, ok := c.configManager.LuaScriptAccessors()[gvk]
-	return ok
+// GetInterpreter returns interpreter for specific resource kind and operation.
+// Why don't support `HookEnabled` function?
+//    When `HookEnabled` tells true, while when methods are calling, configure may be removed, and panic.
+func (c *ConfigurableInterpreter) GetInterpreter(kind schema.GroupVersionKind, operation configv1alpha1.InterpreterOperation) (engine.Function, bool) {
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	interpreter, ok := c.interpreters[kind][operation]
+	return interpreter, ok
 }
 
-// GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
-func (c *ConfigurableInterpreter) GetReplicas(object *unstructured.Unstructured) (int32, *workv1alpha2.ReplicaRequirements, error) {
-	klog.Infof("ConfigurableInterpreter Execute ReviseReplica")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetReplicaResourceLuaScript()) == 0 {
-		return 0, nil, fmt.Errorf("customized interpreter operation GetReplicas  for %q not found", object.GroupVersionKind())
+func (c *ConfigurableInterpreter) updateConfiguration() {
+	objs, err := c.configLister.List(labels.Everything())
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("error updating configuration: %v", err))
+		return
+	}
+	configs := make([]*configv1alpha1.ResourceInterpreterCustomization, len(objs))
+	for i, obj := range objs {
+		config := &configv1alpha1.ResourceInterpreterCustomization{}
+		if err = helper.ConvertToTypedObject(obj, config); err != nil {
+			klog.Errorf("Failed to transform ResourceInterpreterCustomization: %w", err)
+			return
+		}
+		configs[i] = config
 	}
 
-	luaScript := customAccessor.GetReplicaResourceLuaScript
-	klog.Infof("lua script %s", luaScript)
+	newInterpreters, err := loadConfig(configs)
+	if err != nil {
+		klog.Error(err)
+	}
 
-	return 0, nil, nil
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	c.interpreters = newInterpreters
 }
 
-// ReviseReplica revises the replica of the given object.
-func (c *ConfigurableInterpreter) ReviseReplica(object *unstructured.Unstructured, replica int64) (*unstructured.Unstructured, error) {
-	klog.Infof("ConfigurableInterpreter Execute ReviseReplica")
-	customAccessor := c.configManager.LuaScriptAccessors()[object.GroupVersionKind()]
-	if len(customAccessor.GetReplicaRevisionLuaScript()) == 0 {
-		return nil, fmt.Errorf("customized interpreter operation ReviseReplica  for %q not found", object.GroupVersionKind())
-	}
-
-	luaScript := customAccessor.GetReplicaRevisionLuaScript()
-	klog.Infof("lua script %s", luaScript)
+// GetReplicas returns the desired replicas of the object as well as the requirements of each replica.
+func GetReplicas(eng engine.Function, object *unstructured.Unstructured) (int32, *workv1alpha2.ReplicaRequirements, error) {
```

Comment:
How about adding a receiver `(c *ConfigurableInterpreter)` for the method? I think it's more convenient for method calls, just like:
```
i.configurableInterpreter.GetInterpreter()
```
And this keeps the format uniform with other interpreters.